feat: add `blas/ext/base/gvander` (#10934)
stdlib-js/stdlib: lib/node_modules/@stdlib/blas/ext/base/gvander/docs/repl.txt

> {{alias}}( order, mode, M, N, x, strideX, out, LDO )

Reviewer: ```suggestion
{{alias}}( order, mode, M, N, x, strideX, out, ldo )
```

Lowercase. Applies here and elsewhere. See, e.g., `blas/base/dgemv`.

    Generates a Vandermonde matrix.

> 

Reviewer: You need to inform the reader the way in which the matrix is constructed. Namely, that rows are increasing/decreasing powers of x. This is likely best described with a visual:

```
Generates a Vandermonde matrix.

When the mode is positive, the matrix is generated such that

    [
        1   x_0^1   x_0^2   ...   x_0^N
        1   x_1^1   x_1^2   ...   x_1^N
        ...
    ]

with increasing powers along the rows.

When the mode is negative, the matrix is generated such that

    [
        ...
    ]

with decreasing powers along the rows.
```

Otherwise, readers have no idea how the data is stored.

    The `M` and stride parameters determine which elements in the strided
    arrays are accessed at runtime.

> 

Reviewer: ```suggestion
```

This comment is muddied by the fact that we have both `x` and `out`. For Level 2 and 3 BLAS, we do not include this note for this reason.

    Indexing is relative to the first index. To introduce an offset, use a
    typed array view.

> 

Reviewer: ```suggestion
```

Similar to above, we typically only include these comments for Level 1 routines.

    If `M <= 0` or `N <= 0`, the function returns the output array unchanged.

    Parameters
    ----------
    order: string

>         Row-major or column-major storage layout.

Reviewer: ```suggestion
        Row-major (C-style) or column-major (Fortran-style) order.
```

You need to be following our description conventions used in Level 2 and 3 BLAS. See those packages.

    mode: integer

>         Mode indicating whether to generate increasing (`1`) or decreasing

Reviewer: Can we take inspiration from `sort` here which reads:

> Sort order. If `order < 0`, the function sorts `x` in decreasing order. If `order > 0`, the function sorts `x` in increasing order.

The implementation currently does not (nor should it) assume strictly `+-1`. Applies here and elsewhere.

        (`-1`) powers.

    M: integer

>         Number of rows in `out`.

Reviewer: You need to mention that this is also the number of indexed elements in `x`.

    N: integer
        Number of columns in `out`.

    x: Array<number>|TypedArray
        Input array.

    strideX: integer
        Stride length for `x`.

    out: Array<number>|TypedArray

>         Output array.

Reviewer: ```suggestion
        Output matrix.
```

Applies here and elsewhere.

    LDO: integer

>         Stride length for the leading dimension of `out`.

Reviewer: ```suggestion
        Stride of the first dimension of `out` (a.k.a., leading dimension of the
        matrix `out`).
```

Follow conventions used elsewhere (i.e., Level 2 and 3 BLAS).

    Returns
    -------
    out: Array<number>|TypedArray

>         Output array.

Reviewer: ```suggestion
        Output matrix.
```

    Examples
    --------
    // Standard Usage:
    > var x = [ 1.0, 2.0, 3.0 ];
    > var out = [ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 ];
    > {{alias}}( 'row-major', 1, 3, 3, x, 1, out, 3 )
    [ 1.0, 1.0, 1.0, 1.0, 2.0, 4.0, 1.0, 3.0, 9.0 ]

    // Decreasing mode:
    > x = [ 1.0, 2.0, 3.0 ];
    > out = [ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 ];
    > {{alias}}( 'row-major', -1, 3, 3, x, 1, out, 3 )
    [ 1.0, 1.0, 1.0, 4.0, 2.0, 1.0, 9.0, 3.0, 1.0 ]

{{alias}}.ndarray( mode, M, N, x, strideX, offsetX, out, so1, so2, oo )
    Generates a Vandermonde matrix using alternative indexing semantics.

    While typed array views mandate a view offset based on the underlying
    buffer, the offset parameters support indexing semantics based on
    starting indices.

    Parameters
    ----------
    mode: integer

>         Mode indicating whether to generate increasing (`1`) or decreasing

Reviewer: All the same comments.
